City-of-Helsinki/drupal-hdbt-admin
# UHF-5007 CKEditor Language selector

# CKEditor Language selector [UHF-5007](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5007)

A better UX language selector plugin for the CKEditor.

## What was done
* Updated webpack configuration to match with HDBT theme
* Added language selector plugin for the CKEditor and styles for the dialog

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the HDBT Admin theme
    * `composer require drupal/hdbt_admin:dev-UHF-5007_ckeditor_language_selector && composer require drupal/helfi_platform_config:dev-UHF-5007_ckeditor_language_selector`
* Run `make drush-cr drush-updb`

## How to test
* Go to edit any page with CKEditor
* In CKEditor
   * Write a few paragraphs of text and add a phrase in some different language
   * Select the different language part of text and click on the Language selector button next to "Code"
   ![image](https://user-images.githubusercontent.com/1712902/161201773-e7645f75-8d35-4893-9881-60492ce6345b.png)
   * Select a language from the dropdown and click ok.
   * The selected part of text should now be in your chosen language
   * Check: https://user-images.githubusercontent.com/1712902/161203000-50706ead-7cd6-41e8-92a5-fb37e2006611.mov
* Save the node (or entity) and check that the translated texts have `<span lang="" dir=""></span>` tags.

## Other PRs
* https://github.com/City-of-Helsinki/drupal-helfi-platform-config/pull/293
